internal/dispatcher/rabbitmq.go

```diff
@@ -0,0 +1,140 @@
+package dispatcher
+
+import (
+	"context"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	amqp "github.com/rabbitmq/amqp091-go"
+)
+
+const (
+	DefaultPublishTimeout = 5 * time.Second
+)
+
+type RabbitMQDispatcher struct {
+	server   string
+	exchange string
+	conn     *amqp.Connection
+	ch       *amqp.Channel
+}
+
+func NewRabbitMQDispatcher(server string, exchange string) *RabbitMQDispatcher {
+	rd := &RabbitMQDispatcher{
+		server:   server,
+		exchange: exchange,
+	}
+
+	conn, err := amqp.Dial(rd.server)
+	failOnError(err, "Failed to connect to RabbitMQ")
+
+	rd.conn = conn
+
+	ch, err := conn.Channel()
+	failOnError(err, "Failed to open a channel")
+
+	rd.ch = ch
+
+	err = ch.ExchangeDeclare(
+		rd.exchange, // exchange name
+		"direct",    // type
+		true,        // durable
+		false,       // auto-deleted
+		false,       // internal
+		false,       // no-wait
+		nil,         // arguments
+	)
+	failOnError(err, "Failed to declare an exchange")
+
+	return rd
+}
+
+func failOnError(err error, msg string) {
+	if err != nil {
+		log.Panicf("%v: %v", msg, err)
+	}
+}
+
+func (rd *RabbitMQDispatcher) Publish(topic string, data []byte) error {
+	ctx, cancelFn := context.WithTimeout(context.Background(), DefaultPublishTimeout)
+	defer cancelFn()
+
+	if err := rd.ch.PublishWithContext(ctx,
+		rd.exchange, // exchange
+		topic,       // routing key
+		false,       // mandatory
+		false,       // immediate
+		amqp.Publishing{
+			ContentType: "text/plain",
+			Body:        data,
+		}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (rd *RabbitMQDispatcher) Subscribe(topic string, handler MsgHandler) error {
+	// Graceful shutdown the subscriber
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
+	defer stop()
+
+	q, err := rd.ch.QueueDeclare(
+		"",    // queue name (use auto-generated name here, might need to change due to the requirement)
+		false, // durable
```

Comment:
We can use a durable, named queue here instead in case we need to make sure the messages in the queue persist when the consumer restarts (if we follow that way, we need to store the fixed queue name in the `rd` instance for the subscriber when NewRabbitMQDispatcher is called)